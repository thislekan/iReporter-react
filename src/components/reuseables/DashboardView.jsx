/* eslint-disable jsx-a11y/no-static-element-interactions */
/* eslint-disable jsx-a11y/interactive-supports-focus */
/* eslint-disable jsx-a11y/click-events-have-key-events */
import React from 'react';
import cx from 'classnames';
import PropTypes from 'prop-types';
import Modal from 'react-modal';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import CreateReport from '../CreateReport.jsx';
import SingleReport from './SingleReport.jsx';
import ReportDetails from './ReportDetails.jsx';
import FilterComponent from '../search-and-filter/FilterComponent.jsx';
import SearchComponent from '../search-and-filter/SearchComponent.jsx';
import style from '../../styles/UserDashboard.css';
import promoImage from '../../media/promotion.svg';
import Loader from './Loader.jsx';
import EditReport from './EditReport.jsx';

const modalDefaultStyle = { content: { left: '20px', right: '20px', margin: '1rem 0' } };
const customizedStyle = {
  content: {
    left: '20px', right: '20px', padding: '10px', margin: '1rem 0',
  },
};

Modal.setAppElement('#app');

function DashboardView(props) {
  const {
    isLoading,
    closeCreateModal,
    handleChange,
    handleCreateIncident,
    disableFileInput,
    handleMedia,
    // videoBlob,
    imageBlobs,
    clearBlobs,
    openCreateModal,
    isDetailsModalOpen,
    closeDetailsModal,
    incidents,
    location,
    isCreateModalOpen,
    hideCreateButton,
    showCreateButton,
    fetchIncident,
    incident: details,
    mapUrl,
    isDeleteModalOpen,
    alterDeleteModal,
    deleteIncident,
    alterEditModal,
    isEditModalOpen,
    alterDisableComment,
    alterDisableLocation,
    disableComment,
    disableLocation,
    comment,
    incidentLocation,
  } = props;
  return (
    <div className={style.user__dashboard}>
      <div className={style['padded-body']}>
        {isLoading && <Loader isLoading={isLoading} />}
        <div>
          <Modal
            style={modalDefaultStyle}
            isOpen={isCreateModalOpen}
            onRequestClose={closeCreateModal}
            onAfterOpen={hideCreateButton}
            onAfterClose={hideCreateButton}
          >
            <div
              className={style.close__modal}
              onClick={closeCreateModal}
            >
              <FontAwesomeIcon
                icon='times-circle'
                style={{ height: '2rem', width: '2rem' }}
              />
            </div>
            <CreateReport
              handleCreateIncident={handleCreateIncident}
              handleChange={handleChange}
              handleMedia={handleMedia}
              disableFileInput={disableFileInput}
              // videoBlob={videoBlob}
              imageBlobs={imageBlobs}
              clearBlobs={clearBlobs}
            />
          </Modal>
          <Modal
            isOpen={isEditModalOpen}
            onRequestClose={alterEditModal}
            style={customizedStyle}
          >
            <div
              className={style.close__modal}
              onClick={alterEditModal}
            >
              <FontAwesomeIcon
                icon='times-circle'
                style={{ height: '2rem', width: '2rem' }}
              />
            </div>
            <EditReport
              incident={details}
              disableComment={disableComment}
              disableLocation={disableLocation}
              alterDisableComment={alterDisableComment}
              alterDisableLocation={alterDisableLocation}
              handleChange={handleChange}
              comment={comment}
              incidentLocation={incidentLocation}
            />
          </Modal>
        </div>
        <p id="welcome-text">Welcome. View your reports.</p>
        {
          (showCreateButton) ? null : <div
            className={style['create-modal']}
            id="create-modal"
            onClick={openCreateModal}
            role='button'
          >
            <img src={promoImage} alt="" />
          </div>}
        <Modal
          isOpen={isDeleteModalOpen}
          onRequestClose={alterDeleteModal}
          className={style['delete-modal-box']}
        >
          <div className={style['delete-prompt']}>
            <div className={style['prompt-body']}>
              <h3>Are you sure you want to delete this report?</h3>
              <div className={style['delete-btn-parent-div']}>
                <div>
                  <button
                    className={cx(style['alert-btns'], style['cancel-btn'])}
                    onClick={alterDeleteModal}
                  >
                    Cancel
                  </button>
                </div>
                <div>
                  <button
                    className={cx(style['delete-btn'], style['alert-btns'])}
                    onClick={deleteIncident}
                  >
                    Delete Report
                  </button>
                </div>
              </div>
            </div>
          </div>
        </Modal>
        <Modal
          style={customizedStyle}
          isOpen={isDetailsModalOpen}
          onRequestClose={closeDetailsModal}
        >
          <div
            className={style.close__modal}
            onClick={closeDetailsModal}
          >
            <FontAwesomeIcon
              icon='times-circle'
              style={{ height: '2rem', width: '2rem' }}
            />
          </div>
          <ReportDetails
            incident={details}
            mapUrl={mapUrl}
          />
        </Modal>
        <div className={style.body__content}>
          <div className={style.control}>
            <FilterComponent />
            <SearchComponent />
          </div>
          <div className="recently-created">
            {/* <h2 id="current-view-title"></h2> */}
            <div id="all-reports">
              <div
                className={style['list-of-reports']}
                id="incidents-view"
              >
                {
                  (!incidents.length) ? <p>
                    No incident reported yet
                      </p> : incidents.map(incident => <SingleReport
                      key={incident.id}
                      location={location}
                      id={incident.id}
                      createdOn={incident.createdOn}
                      status={incident.status}
                      type={incident.type}
                      incidentLocation={incident.location}
                      fetchIncident={fetchIncident}
                      alterDeleteModal={alterDeleteModal}
                      title={incident.title}
                      alterEditModal={alterEditModal}
                    />)
                }
              </div>
            </div>
          </div>
        </div>
      </div>
    </div >
  );
}

DashboardView.propTypes = {
  isLoading: PropTypes.bool.isRequired,
  closeCreateModal: PropTypes.func.isRequired,
  handleChange: PropTypes.func.isRequired,
  handleCreateIncident: PropTypes.func.isRequired,
  disableFileInput: PropTypes.bool.isRequired,
  handleMedia: PropTypes.func.isRequired,
  // videoBlob: PropTypes.array,
  imageBlobs: PropTypes.array,
  clearBlobs: PropTypes.func.isRequired,
  openCreateModal: PropTypes.func.isRequired,
  isDetailsModalOpen: PropTypes.bool.isRequired,
  closeDetailsModal: PropTypes.func.isRequired,
  incidents: PropTypes.array.isRequired,
  location: PropTypes.object.isRequired,
  isCreateModalOpen: PropTypes.bool.isRequired,
  hideCreateButton: PropTypes.func.isRequired,
  fetchIncident: PropTypes.func.isRequired,
  incident: PropTypes.object.isRequired,
  mapUrl: PropTypes.string,
  showCreateButton: PropTypes.bool.isRequired,
  isDeleteModalOpen: PropTypes.bool.isRequired,
  alterDeleteModal: PropTypes.func.isRequired,
  deleteIncident: PropTypes.func.isRequired,
  alterEditModal: PropTypes.func.isRequired,
  isEditModalOpen: PropTypes.bool.isRequired,
  alterDisableComment: PropTypes.func,
  alterDisableLocation: PropTypes.func,
  disableComment: PropTypes.bool,
  disableLocation: PropTypes.bool,
  comment: PropTypes.string,
  incidentLocation: PropTypes.string,
};

DashboardView.defaultProps = {
  mapUrl: '',
  alterDisableComment: f => f,
  alterDisableLocation: f => f,
  disableComment: true,
  disableLocation: true,
  incidentLocation: '',
  comment: '',
};

export default DashboardView;
