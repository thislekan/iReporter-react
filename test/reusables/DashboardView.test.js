import React from 'react';
import { shallow } from 'enzyme';
import DashboardView from '../../src/components/reuseables/DashboardView.jsx';
import incident from '../mock/incident';

const mock = incident;
const mockFn = jest.fn();

describe('<DashboardView />', () => {
  it('should render DashboardView', () => {
    const wrapper = shallow(
      <DashboardView
        isLoading={false}
        isDetailsModalOpen={false}
        incidents={[mock]}
        location={{ pathname: '/user' }}
        handleChange={mockFn}
        closeDetailsModal={mockFn}
        fetchIncident={mockFn}
        incident={mock}
        imageBlobs={mock.images}
        deleteIncident={mockFn}
        alterDeleteModal={mockFn}
        resetState={mockFn}
        message=""
        openModal={false}
      />,
    );
    // console.log(wrapper.debug());
    expect(wrapper).toMatchSnapshot();
  });
});
