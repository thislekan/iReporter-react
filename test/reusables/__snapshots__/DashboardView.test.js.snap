// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DashboardView /> should render DashboardView 1`] = `
<div>
  <div>
    <div>
      <Modal
        ariaHideApp={true}
        bodyOpenClassName="ReactModal__Body--open"
        closeTimeoutMS={0}
        isOpen={false}
        onAfterClose={[Function]}
        onAfterOpen={[Function]}
        onRequestClose={[Function]}
        parentSelector={[Function]}
        portalClassName="ReactModalPortal"
        role="dialog"
        shouldCloseOnEsc={true}
        shouldCloseOnOverlayClick={true}
        shouldFocusAfterRender={true}
        shouldReturnFocusAfterClose={true}
      >
        <div
          aria-hidden={true}
          onClick={[Function]}
        >
          <FontAwesomeIcon
            border={false}
            className=""
            fixedWidth={false}
            flip={null}
            icon="times-circle"
            inverse={false}
            listItem={false}
            mask={null}
            pull={null}
            pulse={false}
            rotation={null}
            size={null}
            spin={false}
            style={
              Object {
                "height": "2rem",
                "width": "2rem",
              }
            }
            symbol={false}
            title=""
            transform={null}
          />
        </div>
        <CreateReport
          clearBlobs={[Function]}
          disableFileInput={false}
          handleChange={[MockFunction]}
          handleCreateIncident={[Function]}
          handleMedia={[Function]}
          imageBlobs={
            Array [
              "hi",
              "helo",
            ]
          }
        />
      </Modal>
      <Modal
        ariaHideApp={true}
        bodyOpenClassName="ReactModal__Body--open"
        closeTimeoutMS={0}
        isOpen={false}
        onRequestClose={[Function]}
        parentSelector={[Function]}
        portalClassName="ReactModalPortal"
        role="dialog"
        shouldCloseOnEsc={true}
        shouldCloseOnOverlayClick={true}
        shouldFocusAfterRender={true}
        shouldReturnFocusAfterClose={true}
      >
        <div
          aria-hidden={true}
          onClick={[Function]}
        >
          <FontAwesomeIcon
            border={false}
            className=""
            fixedWidth={false}
            flip={null}
            icon="times-circle"
            inverse={false}
            listItem={false}
            mask={null}
            pull={null}
            pulse={false}
            rotation={null}
            size={null}
            spin={false}
            style={
              Object {
                "height": "2rem",
                "width": "2rem",
              }
            }
            symbol={false}
            title=""
            transform={null}
          />
        </div>
        <EditReport
          alterDisableComment={[Function]}
          alterDisableLocation={[Function]}
          alterDisableStatus={[Function]}
          comment=""
          disableComment={true}
          disableLocation={true}
          disableStatus={true}
          handleChange={[MockFunction]}
          incident={
            Object {
              "comment": "jhdfhfdjk",
              "createdBy": "hjdjhdf",
              "createdOn": "48595940698459540",
              "creator": "jhdfkjdf",
              "id": "djfjfdkl-jeroier",
              "images": Array [
                "hi",
                "helo",
              ],
              "isLoading": false,
              "location": "sdjnkjs",
              "message": "kjfdj",
              "status": "draft",
              "title": "dhjfd",
              "type": "red-flag",
              "updatedOn": "19234895049",
              "videos": Array [
                "fdkfdn",
              ],
            }
          }
          incidentLocation=""
          incidentStatus=""
          location={
            Object {
              "pathname": "/user",
            }
          }
        />
      </Modal>
    </div>
    <p
      id="welcome-text"
    >
      Welcome. View your reports.
    </p>
    <div
      aria-hidden={true}
      id="create-modal"
      onClick={[Function]}
      role="button"
    >
      <img
        alt=""
        src={Object {}}
      />
    </div>
    <Modal
      ariaHideApp={true}
      bodyOpenClassName="ReactModal__Body--open"
      closeTimeoutMS={0}
      isOpen={false}
      onRequestClose={[MockFunction]}
      parentSelector={[Function]}
      portalClassName="ReactModalPortal"
      role="dialog"
      shouldCloseOnEsc={true}
      shouldCloseOnOverlayClick={true}
      shouldFocusAfterRender={true}
      shouldReturnFocusAfterClose={true}
    >
      <div>
        <div>
          <h3>
            Are you sure you want to delete this report?
          </h3>
          <div>
            <div>
              <button
                className=""
                onClick={[MockFunction]}
              >
                Cancel
              </button>
            </div>
            <div>
              <button
                className=""
                onClick={[MockFunction]}
              >
                Delete Report
              </button>
            </div>
          </div>
        </div>
      </div>
    </Modal>
    <Modal
      ariaHideApp={true}
      bodyOpenClassName="ReactModal__Body--open"
      closeTimeoutMS={0}
      isOpen={false}
      onRequestClose={[MockFunction]}
      parentSelector={[Function]}
      portalClassName="ReactModalPortal"
      role="dialog"
      shouldCloseOnEsc={true}
      shouldCloseOnOverlayClick={true}
      shouldFocusAfterRender={true}
      shouldReturnFocusAfterClose={true}
    >
      <div
        aria-hidden={true}
        onClick={[MockFunction]}
      >
        <FontAwesomeIcon
          border={false}
          className=""
          fixedWidth={false}
          flip={null}
          icon="times-circle"
          inverse={false}
          listItem={false}
          mask={null}
          pull={null}
          pulse={false}
          rotation={null}
          size={null}
          spin={false}
          style={
            Object {
              "height": "2rem",
              "width": "2rem",
            }
          }
          symbol={false}
          title=""
          transform={null}
        />
      </div>
      <ReportDetails
        incident={
          Object {
            "comment": "jhdfhfdjk",
            "createdBy": "hjdjhdf",
            "createdOn": "48595940698459540",
            "creator": "jhdfkjdf",
            "id": "djfjfdkl-jeroier",
            "images": Array [
              "hi",
              "helo",
            ],
            "isLoading": false,
            "location": "sdjnkjs",
            "message": "kjfdj",
            "status": "draft",
            "title": "dhjfd",
            "type": "red-flag",
            "updatedOn": "19234895049",
            "videos": Array [
              "fdkfdn",
            ],
          }
        }
        mapUrl=""
      />
    </Modal>
    <div>
      <div>
        <FilterFeature />
        <SearchFeature />
      </div>
      <div
        className="recently-created"
      >
        <div
          id="all-reports"
        >
          <div
            id="incidents-view"
          >
            <SingleReport
              alterDeleteModal={[MockFunction]}
              alterEditModal={[Function]}
              createdOn="48595940698459540"
              fetchIncident={[MockFunction]}
              id="djfjfdkl-jeroier"
              incidentLocation="sdjnkjs"
              key="djfjfdkl-jeroier"
              location={
                Object {
                  "pathname": "/user",
                }
              }
              status="draft"
              title="dhjfd"
              type="red-flag"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
